### PR TITLE
Add beginnings of support for CUDA device functions

### PR DESCRIPTION
**Pull request by kernhanda**

### CODE REVIEW

File: hatlib/cuda_loader.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+#define M_PI 3.14159265358979323846
+// Note: Global namespace already includes CUDA math funcs
+//using namespace __jitify_math_ns;
+""",
+    'cuda_fp16.h': "",
+}
+
+HEADER_MAP['climits'] = HEADER_MAP['limits.h']
+
+PROG_NAME = "test_gpu_vec_add"
```
Comment (lisaong) [inline hatlib/cuda_loader.py:238]: curious if these are placeholder names or something

File: hatlib/cuda_loader.py [hunk 2]
```diff
@@ -0,0 +1,392 @@
+import os
+import pathlib
+import sys
+import numpy as np
+from functools import reduce
+from typing import Dict, List
+
+# CUDA stuff
+# TODO: move from pvnrtc module to cuda entirely to reduce dependencies
+from pynvrtc.compiler import Program
```
Comment (lisaong) [inline hatlib/cuda_loader.py:10]: Assuming we still have a dependency later on, will be good to prompt the user to install this at runtime

Alternatively, if there's platform detection, we can do conditional requires like this in setup.cfg:

    vswhere; sys_platform == "win32"

**Review by lisaong — APPROVED** [review]
lgtm!

File: hatlib/cuda_loader.py (hunk 1 above)
Comment (kernhanda) [reply hatlib/cuda_loader.py:238]: the `PROG_NAME` part is from prototyping. The `HEADER_MAP` assignment is basically using the same header body for two different headers (`climits` and `limits.h`)

File: hatlib/cuda_loader.py (hunk 2 above)
Comment (kernhanda) [reply hatlib/cuda_loader.py:10]: Yup, absolutely. I'll add suggestions for user action for missing packages in an upcoming PR

**Pull request merged**